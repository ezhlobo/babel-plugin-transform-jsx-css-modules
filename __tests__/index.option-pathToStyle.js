const pluginTester = require('babel-plugin-tester')
const plugin = require('../index.js')

pluginTester({
  plugin,
  pluginOptions: {
    pathToStyles: /^\.\/module.scss$/,
  },
  title: 'with options: "pathToStyles"',
  tests: [
    {
      title: 'does nothing with default import',
      code: `
        import './styles.css';

        <div className="global"></div>;
      `,
    },

    {
      title: 'amends class names if styles import is provided',
      snapshot: true,
      code: `
        import './module.scss';

        <div className="global-one global-two" styleName="local-one local-two"></div>;
      `,
    },
  ],
})
