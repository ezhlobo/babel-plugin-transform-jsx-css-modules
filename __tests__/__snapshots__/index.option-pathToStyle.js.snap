// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with options: "pathToStyles" amends class names if styles import is provided: amends class names if styles import is provided 1`] = `
"
import './module.scss';

<div className=\\"global-one global-two\\" styleName=\\"local-one local-two\\"></div>;

      ↓ ↓ ↓ ↓ ↓ ↓

import __CSSM__ from \\"./module.scss\\";


<div className={[\\"global-one\\", \\"global-two\\", __CSSM__[\\"local-one\\"], __CSSM__[\\"local-two\\"]].join(\\" \\")}></div>;
"
`;
